crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java

```diff
@@ -44,7 +45,7 @@ private CryptoStreamFactory() {}
      * cipher {@code algorithm}. When OpenSSL is available an implementation that uses AES-NI will be returned.
      */
     public static SeekableInput decrypt(SeekableInput encryptedInput, KeyMaterial keyMaterial, String algorithm) {
-        if (!algorithm.equals(AES_ALGORITHM)) {
+        if (!algorithm.equals(AES_ALGORITHM) || Boolean.parseBoolean(System.getProperty(FORCE_JCE))) {
```

Comment:
this seems like an orthogonal change? prefer 2 PRs, one to fix the bug and one to allow overriding JCE